react/src/components/Preview.js

```diff
@@ -1,7 +1,32 @@
-import React from 'react'
-import { markdown } from 'markdown'
+import React, { Component } from 'react'
+import { loadPreview } from './Api'
+// import './css/pdf.css'
+
 /* eslint react/no-danger: 0 */
-const Preview = props =>
-  <div dangerouslySetInnerHTML={{ __html: markdown.toHTML(props.text) }} />
+class Preview extends Component {
+  state = {
+    html: '',
+  }
+
+  componentDidMount() {
+    this.updatePreview()
+  }
+
+  componentWillReceiveProps(props) {
+    this.updatePreview(props.text)
+  }
```

Comment:
Tätä metodia kutsutaan useasti kun propsit muuttuu (minkä vuoksi /actions/pdf POST requesteja lähtee monta). Saattaa liittyä jotenkin tähän: https://medium.com/@nimelrian/as-of-react-16-you-should-use-componentwillreceiveprops-only-to-update-state-synchronously-as-in-a9d66457c51 . Varmin/helpoin tapa korjata tää olis tehdä komponentista stateless ja html tulee propsina, mut en tiedä onko se paras tapa.